packages/wallets/src/components/Base/Badge/Badge.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { Children, cloneElement, FC, ReactElement, ReactNode } from 'react';
+import classNames from 'classnames';
+import { TGenericSizes } from '../../../types';
+import { WalletText } from '../WalletText';
+import { WalletTextProps } from '../WalletText/WalletText';
+import './Badge.scss';
+
+interface BadgeComponent extends FC<BadgeProps> {
+    Icon: React.FC<BadgeIconProps>;
+    Text: React.FC<BadgeTextProps>;
+}
+interface BadgeProps {
+    children: ReactElement | ReactElement[];
+    className?: string;
+    size: Extract<TGenericSizes, 'lg' | 'md' | 'sm' | 'xs'>; // TODO: add css classes to handle the size of the badges and map it to the text sizes.
+}
+
+interface BadgeTextProps {
+    children: WalletTextProps['children'];
+    color?: WalletTextProps['color'];
+    isBold?: boolean;
+    size?: WalletTextProps['size'];
+}
```

Comment:
Cant we just extend WalletTextProps and combine BadgeTextProps and BadgeIconProps in one ?